test/GSNProvider.test.js

```diff
@@ -518,6 +519,59 @@ describe('GSNProvider', function () {
     });
 
   });
+
+  context('detecting relays added', function () {
+    before('getting relayHub', async function () {
+      this.relayHub = await getRelayHub(this.web3);
+    });
+
+    beforeEach('creating provider', async function () {
+      this.gsnProvider = new GSNProvider(PROVIDER_URL, HARDCODED_RELAYER_OPTS);
+      this.greeter.setProvider(this.gsnProvider);
+      await this.greeter.methods.greet("Hello").send({ from: this.sender, useGSN: true });
+    });
+
+    it('finds the one relay added', async function () {
+      const relays = await this.gsnProvider.relayClient.serverHelper.fetchRelaysAdded();
+      expect(relays).to.have.lengthOf(1);
+    });
+
+    context('with a second relay added', async function () {
+      beforeEach('adding second relay', async function () {
+        await this.relayHub.methods.stake(this.secondRelay, 60*60*24*10 /* 10 days */)
+          .send({
+            value: Web3.utils.toWei('1', 'ether'),
+            from: this.deployer,
+            useGSN: false,
```

Comment:
`this.relayHub` should be connected to a vanilla provider, so no need for the useGSN flag here

Reply:
Are you okay with leaving it just for explicitness?

Reply:
Definitely